### ISSUE TITLE
On MacOS SysTray menu does not show when clicked on an active space in second monitor

### ISSUE THREAD

**Issue by udelledo** [fyne-io/fyne#5223]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The basic sample application found on the documentation when running on a MacOS shows the systray menu on when the systray icon is clicked in a _Desktop_ window, it doesn't show anything when it's clicked with a [space](https://support.apple.com/en-gb/guide/mac-help/mh14112/mac) active

### How to reproduce

1. Run the sample systray application
2. Get to a space on a second monitor (full window application)
3. click on the systray icon
4. The menu does not show

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/driver/desktop"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("SysTray")

	if desk, ok := a.(desktop.App); ok {
		m := fyne.NewMenu("MyApp",
			fyne.NewMenuItem("Show", func() {
				w.Show()
			}))
		desk.SetSystemTrayMenu(m)
	}

	w.SetContent(widget.NewLabel("Fyne System Tray"))
	w.SetCloseIntercept(func() {
		w.Hide()
	})
	w.ShowAndRun()
}
```

### Fyne version

2.5.1

### Go compiler version

1.23.2

### Operating system and version

Mac OS 14.5

**Comment by andydotxyz** [comment]
I cannot replicate this when the primary monitor is the fullscreen app - do you see that too?
I will go find another monitor to plug in.

**Comment by udelledo** [comment]
Yes, I noticed that on the primary monitory it does not happen, but I also just noticed that the fyne app is not the only one that is showing this behavior. Cisco any connect is behaving the same, but Outlook or Oracle content are working correctly.

**Comment by andydotxyz** [comment]
Fixed by using the right macOS fullscreen mode from v2.6.0 when it releases (on develop branch at the moment)
